src/OrchardCore.Modules/OrchardCore.Lucene/Services/LuceneIndexInitializerService.cs

```diff
@@ -11,42 +10,32 @@ namespace OrchardCore.Lucene
     public class LuceneIndexInitializerService : IModularTenantEvents
     {
         private readonly ShellSettings _shellSettings;
-        private readonly ILogger _logger;
-        private readonly LuceneIndexSettingsService _luceneIndexSettingsService;
-        private readonly LuceneIndexingService _luceneIndexingService;
 
-        public LuceneIndexInitializerService(
-            LuceneIndexSettingsService luceneIndexSettingsService,
-            LuceneIndexingService luceneIndexingService,
-            ShellSettings shellSettings,
-            ILogger<LuceneIndexInitializerService> logger)
+        public LuceneIndexInitializerService(ShellSettings shellSettings)
         {
-            _luceneIndexSettingsService = luceneIndexSettingsService;
-            _luceneIndexingService = luceneIndexingService;
             _shellSettings = shellSettings;
-            _logger = logger;
         }
 
         public Task ActivatedAsync()
         {
-            if (_shellSettings.State != TenantState.Uninitialized)
+            if (_shellSettings.State == TenantState.Running)
             {
                 ShellScope.AddDeferredTask(async scope =>
                 {
-                    try
-                    {
-                        var luceneIndexSettings = await _luceneIndexSettingsService.GetSettingsAsync();
+                    var luceneIndexSettingsService = scope.ServiceProvider.GetRequiredService<LuceneIndexSettingsService>();
+                    var luceneIndexingService = scope.ServiceProvider.GetRequiredService<LuceneIndexingService>();
+                    var indexManager = scope.ServiceProvider.GetRequiredService<LuceneIndexManager>();
```

Comment:
Makes sense using the current scope and resolve it's services. +1

Reply:
@Skrypt 

Yes, because otherwise we resolve services not from the right scope service provider

Didn't see it when reviewing, while mounting my SSD ;)

Reply:
Yeah I thought that since these are Singletons that there would be only one instance of them though. But actually there is one of them per scope.